### ISSUE TITLE
dotnet-install.cmd fails if install path contains a space

### ISSUE THREAD

**Issue by johnbeisner** [dotnet/arcade#3941]
From: https://github.com/dotnet/cli/issues/11521

@stephenweatherford

## Steps to reproduce
On Windows, run this:
```
dotnet-install.cmd -InstallDir "c:\Users\<user>\AppData\Roaming\Code - Insiders\User\globalStorage\ms-vscode.sample-extension\.dotnet\2.2.0" -Runtime dotnet -Version 2.2.0
```

## Expected  behavior
Should install

## Actual behavior
Gives an error:
```
Architecture not supported. If you think this is a bug, please report it at https://github.com/dotnet/cli/issues
```

The path is correctly interpreted by the batch file, but PowerShell is getting a corrupted value for the $Archtecture parameter (`"Insiders\User\globalStorage\ms-vscode.sample-extension\.dotnet\2.2.0"`)

## Environment data
`dotnet --info` output:

.NET Core SDK (reflecting any global.json):
 Version:   3.0.100-preview3-010431
 Commit:    d72abce213

Runtime Environment:
 OS Name:     Windows
 OS Version:  10.0.18362
 OS Platform: Windows
 RID:         win10-x64
 Base Path:   C:\Program Files\dotnet\sdk\3.0.100-preview3-010431\

Host (useful for support):
  Version: 3.0.0-preview3-27503-5
  Commit:  3844df9537

.NET Core SDKs installed:
  2.1.403 [C:\Program Files\dotnet\sdk]
  2.1.505 [C:\Program Files\dotnet\sdk]
  2.1.507 [C:\Program Files\dotnet\sdk]
  2.1.602 [C:\Program Files\dotnet\sdk]
  2.1.700-preview-009597 [C:\Program Files\dotnet\sdk]
  2.1.700-preview-009601 [C:\Program Files\dotnet\sdk]
  2.1.800-preview-009677 [C:\Program Files\dotnet\sdk]
  2.2.105 [C:\Program Files\dotnet\sdk]
  3.0.100-preview3-010431 [C:\Program Files\dotnet\sdk]

.NET Core runtimes installed:
  Microsoft.AspNetCore.All 2.1.5 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.All]
  Microsoft.AspNetCore.All 2.1.9 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.All]
  Microsoft.AspNetCore.All 2.1.11 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.All]
  Microsoft.AspNetCore.All 2.2.3 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.All]
  Microsoft.AspNetCore.App 2.1.5 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 2.1.9 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 2.1.11 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 2.2.3 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 3.0.0-preview3-19153-02 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 2.1.5 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 2.1.9 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 2.1.11 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 2.2.3 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 3.0.0-preview3-27503-5 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.WindowsDesktop.App 3.0.0-preview3-27504-2 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]

To install additional .NET Core runtimes or SDKs:
  https://aka.ms/dotnet-download

**Comment by johnbeisner** [comment]
I could not reproduce the failure. I used the following command:
.\dotnet-install.ps1 -Version "2.1.701" -InstallDir "D:\G\Test - Folder" -Verbose "True"

**Comment by johnbeisner** [comment]
This call to powershell is removing all double-quotes in a command:
`powershell -ExecutionPolicy ByPass -NoProfile -command "& """%~dp0dotnet-install.ps1""" %*"`

Example:
D:\G\cli\scripts\obtain>powershell -ExecutionPolicy ByPass -NoProfile -command "& """D:\G\cli\scripts\obtain\dotnet-install.ps1""" -Version "2.1.701" -InstallDir "D:\G\Test - Folder" -Verbose True -DryRun"
```
D:\G\cli\scripts\obtain\dotnet-install.ps1 : Cannot validate argument on parameter 'Runtime'. The argument "True" does not belong to the set "dotnet,aspnetcore,windowsdesktop" specified by the ValidateSet attribute.
Supply an argument that is in the set and then try the command again.
At line:1 char:105
+ ... -Version 2.1.701 -InstallDir D:\G\Test - Folder -Verbose True -DryRun
+                                                              ~~~~
    + CategoryInfo          : InvalidData: (:) [dotnet-install.ps1], ParameterBindingValidationException
    + FullyQualifiedErrorId : ParameterArgumentValidationError,dotnet-install.ps1

```

**Comment by garath** [comment]
So I have some more context on scope, did this come up during "normal" use of Arcade, or by some other path (like calling directly, outside of Arcade)?

**Comment by StephenWeatherford** [comment]
Don't know what arcade is.  This is a direct usage of the scripts from https://docs.microsoft.com/en-us/dotnet/core/tools/dotnet-install-script inside a vscode extension (https://github.com/Microsoft/vscode-azurearmtools).

**Comment by StephenWeatherford** [comment]
BTW, we worked around this by change the double quotes to single quotes, although it seems like either should work, but I guess that's up to you.

**Comment by johnbeisner** [comment]
@StephenWeatherford 
If you are not referring to the 'Arcade' "dotnet-install.cmd"; which "dotnet-install.cmd" is causing this error?

**Comment by johnbeisner** [comment]
@garath 
As a side-note; the 'Arcade' "dotnet-install.cmd" has exactly this problem - I tested it manually and was able to reproduce what Stephen was seeing. You should consider Stephan's solution.

**Comment by StephenWeatherford** [comment]
@johnbeisner I'm not the original author, don't know where they picked it up, I'd assumed https://docs.microsoft.com/en-us/dotnet/core/tools/dotnet-install-script but it looks like that one only has the .ps1/.sh files.

**Comment by johnbeisner** [comment]
@StephenWeatherford
To be clear, the error is not in the .ps1/.sh files - it is in the cmd file: 'dotnet-install.cmd'

**Comment by riarenas** [comment]
Adding to FR so someone can take a look.

The problematic line is https://github.com/dotnet/arcade/blob/b9d0c899d9c80cbf76774ca2792cdef69a60deb6/eng/common/dotnet-install.cmd#L2

**Comment by MattGal** [comment]
The Arcade version of this script (which I admit, has the same name) does not even include an InstallDir path.  I was able to get a repro by Frankensteining together the script from arcade and the PS1 from the download link and am investigating.

**Comment by MattGal** [comment]
OK this is broken in more ways than I'd like, but the main challenge here is just the paths.

- The ps1 file you're using to get this error is from https://github.com/dotnet/cli/blob/master/scripts/obtain/dotnet-install.ps1 .  A quick review of [the folder in the repo](https://github.com/dotnet/cli/tree/master/scripts/obtain) shows us it was never meant to be called by that .cmd file.

- The CMD file you're calling IS indeed the arcade version... which can't be messed up this way because it has far fewer flags and doesn't even allow specifying a location directly: https://github.com/dotnet/cli/blob/master/eng/common/dotnet-install.ps1 and https://github.com/dotnet/cli/blob/master/eng/common/dotnet-install.cmd.

TL;DR: This is an unsupported combination which noone is even claiming ever worked;  if ti works after tweaking some " to ' it's because the cmd wraps stuff in escaped quotes so not surprising.
